src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf16Utility.Validation.cs

```diff
@@ -152,9 +186,16 @@ static Utf16Utility()
                         // Surrogates need to be special-cased for two reasons: (a) we need
                         // to account for the fact that we over-counted in the addition above;
                         // and (b) they require separate validation.
-
-                        utf16Data = Sse2.Add(utf16Data, vectorA800);
-                        mask = (uint)Sse2.MoveMask(Sse2.CompareLessThan(utf16Data.AsInt16(), vector8800).AsByte());
+                        if (AdvSimd.Arm64.IsSupported)
+                        {
+                            utf16Data = AdvSimd.Add(utf16Data, vectorA800);
+                            mask = (uint)Arm64MoveMask(AdvSimd.Arm64.CompareLessThan(utf16Data.AsDouble(), vector8800.AsDouble()).AsByte());
```

Comment:
This `CompareLessThan` method is receiving two doubles, so I'm not entirely certain that it will do the same as the `Sse2` version.

Reply:
There is variance of `CompareLessThan` that takes `short` - [https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/AdvSimd.cs#L5556](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Arm/AdvSimd.cs#L5556)

---
In reply to: [452954531](https://github.com/dotnet/runtime/pull/39050#discussion_r452954531) [](ancestors = 452954531)